Sources/Basics/ProgressAnimation/NinjaProgressAnimation.swift

```diff
@@ -0,0 +1,101 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import class TSCBasic.TerminalController
+import protocol TSCBasic.WritableByteStream
+
+extension ProgressAnimation {
+    /// A ninja-like progress animation that adapts to the provided output stream.
+    @_spi(SwiftPMInternal_ProgressAnimation)
```

Comment:
Not sure it's worth introducing a new `@_spi` namespace for this in addition to existing `@_spi(SwiftPMInternal)`? We're going to replace these with `package` in the future anyway, but in the meantime a single `SwiftPMInternal` namespace seems more preferable to me.